docs/integrations/engines/snowflake.md

```diff
@@ -1,5 +1,296 @@
+    GRANT CREATE TABLE ON FUTURE SCHEMAS IN DATABASE demo_db TO ROLE sqlmesh; -- Can create TABLEs in schemas
+    GRANT CREATE VIEW ON FUTURE SCHEMAS IN DATABASE demo_db TO ROLE sqlmesh; -- Can create VIEWs in schemas
+    GRANT SELECT, INSERT, TRUNCATE, UPDATE, DELETE ON FUTURE TABLES IN DATABASE demo_db TO ROLE sqlmesh; -- Can SELECT and modify TABLEs in schemas
+    GRANT REFERENCES, SELECT ON FUTURE VIEWS IN DATABASE demo_db TO ROLE sqlmesh; -- Can SELECT and modify VIEWs in schemas
+
+    GRANT ROLE sqlmesh TO USER demo_user; -- Grant role to user
+    ALTER USER demo_user SET DEFAULT ROLE = sqlmesh; -- Make role user's default role
+    ```
+
+### Get connection info
+
+Now that our user has sufficient access permissions, we're ready to gather the information needed to configure the SQLMesh connection.
+
+#### Account name
+
+Snowflake connection configurations require the `account` parameter that identifies the Snowflake account SQLMesh should connect to.
+
+Snowflake account identifiers have two components: your organization name and your account name. Both are embedded in your Snowflake web interface URL, separated by a `/`.
+
+This shows the default view when you log in to your Snowflake account, where we can see the two components of the account identifier:
+
+![Snowflake account info in web URL](./snowflake/snowflake_db-guide_account-url.png){ loading=lazy }
+
+In this example, our organization name is `idapznw`, and our account name is `wq29399`.
+
+We concatenate the two components, separated by a `-`, for the SQLMesh `account` parameter: `idapznw-wq29399`.
+
+#### Warehouse name
+
+Your Snowflake account may have more than one warehouse available - any will work for this quickstart, which runs very few computations.
+
+Some Snowflake user accounts may have a default warehouse they automatically use when connecting.
+
+The connection configuration's `warehouse` parameter is not required, but we recommend specifying the warehouse explicitly in the configuration to ensure SQLMesh's behavior doesn't change if the user's default warehouse changes.
+
+#### Database name
+
+Snowflake user accounts may have a "Default Namespace" that includes a default database they automatically use when connecting.
+
+The connection configuration's `database` parameter is not required, but we recommend specifying the database explicitly in the configuration to ensure SQLMesh's behavior doesn't change if the user's default namespace changes.
+
+### Configure the connection
+
+We now have the information we need to configure SQLMesh's connection to Snowflake.
+
+We begin by adding a new gateway named `snowflake` to our example project's config.yaml file:
+
+```yaml linenums="1" hl_lines="6-8"
+gateways:
+  local:
+    connection:
+      type: duckdb
+      database: db.db
+  snowflake:
+    connection:
+      type: snowflake
+
+default_gateway: local
+
+model_defaults:
+  dialect: duckdb
```

Comment:
Should the example default dialect be `dialect: snowflake`?

Otherwise it feels like we are recommending cross-dialect projects in the first instance which have a tendency to trigger all the Snowflake normalization edge cases

Reply:
I had the same thought, but:
- Since we have both the DuckDB and snowflake gateways present, cross-dialect execution is implied either way
- I don't want to show changing the default dialect when we change engines because that's explicitly wrong (in the sense that the dialect of the models is constant regardless of the engine you use)

What do you think about me deleting the DuckDB gateway first thing so we can set the model dialect to snowflake without implying cross-database usage? I think we'd probably still want a brief note about what default dialect is for/how it works?

Reply:
Yeah either that or ignore duckdb entirely in the Snowflake quickstart (to keep the Snowflake quickstart focused on just Snowflake) and just `sqlmesh init snowflake` in the Prerequisites section instead of `sqlmesh init duckdb`